controllers/common_test.go

```diff
@@ -44,63 +43,64 @@ var _ = Describe("GetGeneratedJobName", func() {
 
 	JustBeforeEach(func() {
 		uidWithoutHyphens = strings.Replace(string(uid), "-", "", -1)
```

Comment:
nit: It is weird that the `uidWithoutHyphens` is still here, since you refactored out the logic that removes hyphens. It will make test easier to understand if you remove it since readers will not have to understand this element -- the function `GetGeneratedResourceName` should work for arbitrary string input.

Reply:
It does, but the idea was to check if the length of required is more than maxLen, the second half of required which contains the uid gets picked. I changed this to not use a generated UID but a hardcoded one. Let me know if you prefer to remove this check altogether.

Reply:
> It does, but the idea was to check if the length of required is more than maxLen, the second half of required which contains the uid gets picked

This test is fine but somewhat imprecise -- if `GetGeneratedResourceName` has room only for `required` (no `optional`), why would our tests only check that the last half of `required` is present?

To make the test better match the documented behavior of the function, we should change:
```go
It("Contains the full UID", func() {
    Expect(generatedResourceName).To(ContainSubstring(requiredPostfix))
})
```
to:
```go
It("Contains the full required string", func() {
    Expect(generatedResourceName).To(ContainSubstring(required))
})
```

The impact of this change pretty small so happy to approve either way.